components/delay/delay.go

```diff
@@ -0,0 +1,65 @@
+package delay
+
+import (
+	"context"
+	"time"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+)
+
+// Delay represents a message's delay.
+// It can be either a delay until a specific time or a delay for a specific duration.
+// The zero value of Delay is a zero delay.
+//
+// IMPORTANT: Delay doesn't work with all Pub/Subs! Using it won't have any effect on Pub/Subs that don't support it.
```

Comment:
I woner if we don't want to put list of supported Pub/Subs for that somewhere? it may be not obvious to find

Reply:
Added docs eaa5ff93814041fc6c37339591c394561d37c79f